.github/workflows/docs_deploy.yml

```diff
@@ -19,19 +19,30 @@ jobs:
         with:
           # We need to fetch the whole history so 'reno' can do its job and we can inspect tags.
           fetch-depth: 0
-
       - uses: actions/setup-python@v5
         name: Install Python
         with:
           # Sync with 'documentationPythonVersion' in 'azure-pipelines.yml'.
           python-version: '3.9'
-
       - name: Install dependencies
         run: tools/install_ubuntu_docs_dependencies.sh
 
+      - name: Determine GitHub branch name
+        run: |
+          # PR workflows set the branch they're merging into.
+          if [ -n "$GITHUB_BASE_REF" ]; then
+            BRANCH_NAME="$GITHUB_BASE_REF"
+          # Tags like 1.0.0 and 1.0.0rc1 should point to their stable branch.
+          elif [[ $GITHUB_REF_NAME =~ ^([0-9]+\.[0-9]+) ]]; then
+            BRANCH_NAME="stable/${BASH_REMATCH[1]}"
+          else
+              BRANCH_NAME="$GITHUB_REF_NAME"
```

Comment:
I know that @jakelishman has strong opinions about using the github actions built-in variables here (see: https://github.com/Qiskit/qiskit/pull/11901 ). But personally I think this is ok; but the concerns about manual builds are valid and I wonder how this works if you want to do a local build for debugging that points to the correct place.

Reply:
Jake didn't want the GitHub actions to live inside `conf.py`. He wasn't opposed to using GitHub Actions built-in variables in general, only where those are set. He suggested at the bottom of #11901 setting the values in the CI files like this.

For local builds, we default to `main`. You can override the value by locally setting `QISKIT_DOCS_GITHUB_BRANCH_NAME=stable/1.0`. I don't think it's worth the complexity to try to do something clever. I'd rather we keep it simple, at least until we hear devs banging on the doors that this is a problem for them and they need auto-detection for local builds.

Reply:
I'll just resolve this to move it out the way. For completeness' sake, Eric's right: I was complaining about using the GitHub environment variables within `docs/conf.py`, especially since the PR I was complaining about was trying to make it work with Azure as well - I just wanted to see the CI-specific environment variables being resolved into Qiskit-specific forms in the CI-specific configuration files, which link CI to Qiskit, which is now what's being done.